### DISCUSSION TITLE
Getting errors after migrating from primefaces 7 to 8 in Exporter and export

### DISCUSSION THREAD

**Discussion by Prashant-Raghuvanshi** [primefaces/community#1710] [PrimeFaces] [answered]
Getting the following errors after upgrading the primefaces version cant find a solution to fix those issues please help.
ERROR MESSAGE
/////The type Exporter cannot be the superclass of CustomPdfWriter; a superclass must be a class//////
public class CustomPdfWriter extends Exporter {
///Logic

	@Override
	**ERROR MESSAGE**
	/////The method export(FacesContext, DataTable, String, boolean, boolean, String, MethodExpression, MethodExpression, ExporterOptions, MethodExpression) of type AnalysisReportPdfWriter must override or implement a supertype method//////
	public void export(FacesContext facesContext, DataTable table, String filename,
			boolean pageOnly, boolean selectionOnly, String encodingType,
			MethodExpression preProcessor, MethodExpression postProcessor, ExporterOptions argument1, MethodExpression argument2) throws IOException{
			////Logic
			}
					**ERROR MESSAGE**
			//////ColumnType cannot be resolved to a variable
			    	addColumnFacets(table, pdfTable, ColumnType.HEADER);

**Comment by Rapster** [comment]
Lot have changed since PF 7, start by inheriting TableExporter instead

**Reply by Prashant-Raghuvanshi** [reply]
Can you please suggest how can i implement it. Can u please share some code so that i can take reference

**Reply by Rapster** [reply, marked as answer]
There you go: https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/java/org/primefaces/component/datatable/export/DataTablePDFExporter.java

or something simpler: 

https://github.com/primefaces/primefaces/blob/master/primefaces-showcase/src/main/java/org/primefaces/showcase/view/data/dataexporter/TextExporter.java
